python/ray/_private/runtime_env/packaging.py

```diff
@@ -135,6 +135,47 @@ def _dir_travel(
         excludes.pop()
 
 
+def _hash_single(
```

Comment:
I don't fully understand this name. What about `_hash_file_content_or_directory_name`? Also please write down what are hashed, in the docstring.